R/check_properties.R

```diff
@@ -5,12 +5,14 @@
 #' expected schema
 #' @param properties a named list of input properties
 #' @param schema a list representation of a schema or sub-schema
+#' @param parent_property the name of the parent property, if any. Necessary when
+#' checking output type `value` properties.
 #' @param call the calling environment
 #'
 #' @return NULL. The functions are called for their side effects.
 #' @noRd
 # Check scalar properties against the schema
-check_properties_scalar <- function(properties, schema,
+check_properties_scalar <- function(properties, schema, parent_property = NULL,
```

Comment:
Introducing this argument to enable us to use it to validate output type `values` properties (which needs to know the parent property